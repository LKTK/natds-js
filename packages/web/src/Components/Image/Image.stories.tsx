import * as React from "react";
import { Meta, Story } from "@storybook/react";
import { Image } from "./Image";
import { ImageProps } from "./Image.props";
import { argTypes } from "./Image.argTypes";

export default {
  argTypes,
  component: Image,
  title: "Components/Image",
} as Meta;

const Template : Story<ImageProps> = (args: ImageProps) => <Image {...args} />;

export const Playground : Story<ImageProps> = Template.bind({});
Playground.args = {
  alt: "Nat Natura holding a cat",
  src: "https://pbs.twimg.com/profile_images/1285932878795333632/Wl95KE_m_400x400.jpg",
};
