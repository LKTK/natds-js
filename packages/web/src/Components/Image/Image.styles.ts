import makeStyles from "@material-ui/core/styles/makeStyles";
import createStyles from "@material-ui/core/styles/createStyles";
import { ImageProps } from "./Image.props";
import { IThemeWeb } from "../../Themes";
import { getSizeFromProp } from "../shared/getSizeFromProp";

export const useStyles = makeStyles((theme: IThemeWeb) => createStyles({
  root: ({
    disableSelection, height, maxHeight, maxWidth, width,
  }: ImageProps) => ({
    height: getSizeFromProp({
      sizePropValue: height,
      themeSizes: theme.sizes,
    }),
    maxHeight: getSizeFromProp({
      sizePropValue: maxHeight,
      themeSizes: theme.sizes,
    }),
    maxWidth: getSizeFromProp({
      sizePropValue: maxWidth,
      themeSizes: theme.sizes,
    }),
    userSelect: disableSelection ? "none" : "auto",
    width: getSizeFromProp({
      sizePropValue: width,
      themeSizes: theme.sizes,
    }),
  }),
}), { name: "NatDSImage" });

export default useStyles;
