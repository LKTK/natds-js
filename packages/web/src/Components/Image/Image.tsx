import * as React from "react";
import useTheme from "@material-ui/core/styles/useTheme";
import clsx from "clsx";
import { ImageProps } from "./Image.props";
import { useStyles } from "./Image.styles";
import { getSizeFromProp } from "../shared/getSizeFromProp";
import { IThemeWeb } from "../../Themes";

export { ImageProps } from "./Image.props";

export const Image = React.forwardRef<HTMLImageElement, ImageProps>(
  (props: ImageProps, ref) => {
    const {
      alt, className, draggable, disableSelection, height, maxHeight, maxWidth, style, width, ...otherProps
    } = props;

    const classes = useStyles({
      alt, disableSelection, height, maxHeight, maxWidth, width,
    });

    const theme : IThemeWeb = useTheme();

    const heightAttribute = getSizeFromProp({
      sizePropValue: height,
      themeSizes: theme.sizes,
    });
    const widthAttribute = getSizeFromProp({
      sizePropValue: width,
      themeSizes: theme.sizes,
    });

    return (
      <img
        alt={alt}
        className={clsx(classes.root, className)}
        draggable={draggable}
        height={heightAttribute === "auto" ? "" : heightAttribute}
        ref={ref}
        style={style}
        width={widthAttribute === "auto" ? "" : widthAttribute }
        {...otherProps}
      />
    );
  },
);

Image.displayName = "Image";

export default Image;
