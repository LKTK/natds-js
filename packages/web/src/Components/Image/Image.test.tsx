import * as React from "react";
import * as TestRenderer from "react-test-renderer";
import Image from "./Image";

describe("Image component", () => {
  let testRenderer : TestRenderer.ReactTestRenderer = {} as TestRenderer.ReactTestRenderer;

  describe("when only required properties are provided", () => {
    it("should render correctly", () => {
      testRenderer = TestRenderer.create(<Image alt={"Image alt text"} />);

      expect(testRenderer).toMatchSnapshot();
    });
  });
  describe("when disableSelection property is explicitly set to true", () => {
    it("should render correctly", () => {
      testRenderer = TestRenderer.create(<Image alt={"Image alt text"} disableSelection={true} />);
      expect(testRenderer).toMatchSnapshot();
    });
  });
  describe("when disableSelection property is explicitly set to false", () => {
    it("should render correctly", () => {
      testRenderer = TestRenderer.create(<Image alt={"Image alt text"} disableSelection={false} />);
      expect(testRenderer).toMatchSnapshot();
    });
  });
  describe("when height is set to auto", () => {
    it("should render correctly", () => {
      testRenderer = TestRenderer.create(<Image alt={"Image alt text"} height={"auto"} />);
      expect(testRenderer).toMatchSnapshot();
    });
  });
  describe("when width is set to auto", () => {
    it("should render correctly", () => {
      testRenderer = TestRenderer.create(<Image alt={"Image alt text"} width={"auto"} />);
      expect(testRenderer).toMatchSnapshot();
    });
  });
});
