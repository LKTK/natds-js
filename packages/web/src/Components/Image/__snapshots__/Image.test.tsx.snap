// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image component when disableSelection property is explicitly set to false should render correctly 1`] = `
<img
  alt="Image alt text"
  className="NatDSImage-root-1 NatDSImage-root-4"
  height=""
  width=""
/>
`;

exports[`Image component when disableSelection property is explicitly set to true should render correctly 1`] = `
<img
  alt="Image alt text"
  className="NatDSImage-root-1 NatDSImage-root-3"
  height=""
  width=""
/>
`;

exports[`Image component when height is set to auto should render correctly 1`] = `
<img
  alt="Image alt text"
  className="NatDSImage-root-1 NatDSImage-root-5"
  height=""
  width=""
/>
`;

exports[`Image component when only required properties are provided should render correctly 1`] = `
<img
  alt="Image alt text"
  className="NatDSImage-root-1 NatDSImage-root-2"
  height=""
  width=""
/>
`;

exports[`Image component when width is set to auto should render correctly 1`] = `
<img
  alt="Image alt text"
  className="NatDSImage-root-1 NatDSImage-root-6"
  height=""
  width=""
/>
`;
